sn_networking/src/event/kad.rs

```diff
@@ -369,83 +385,116 @@ impl SwarmDriver {
 
             // Insert the record and the peer into the result_map.
             let record_content_hash = XorName::from_content(&peer_record.record.value);
-            let responded_peers =
+
+            let peer_list =
                 if let Entry::Occupied(mut entry) = result_map.entry(record_content_hash) {
                     let (_, peer_list) = entry.get_mut();
+
                     let _ = peer_list.insert(peer_id);
-                    peer_list.len()
+                    peer_list.clone()
                 } else {
                     let mut peer_list = HashSet::new();
                     let _ = peer_list.insert(peer_id);
-                    result_map.insert(record_content_hash, (peer_record.record.clone(), peer_list));
-                    1
+                    result_map.insert(
+                        record_content_hash,
+                        (peer_record.record.clone(), peer_list.clone()),
+                    );
+
+                    peer_list
                 };
 
+            let responded_peers = peer_list.len();
+
             let expected_answers = get_quorum_value(&cfg.get_quorum);
+            trace!("Expecting {expected_answers:?} answers to exceed {expected_get_range:?} for record {pretty_key:?} task {query_id:?}, received {responded_peers} so far");
+
+            let data_key_address = NetworkAddress::from_record_key(&key);
+            let is_sensitive_data = cfg.get_quorum == Quorum::All;
+
+            let we_have_searched_thoroughly = Self::have_we_have_searched_thoroughly_for_quorum(
+                expected_get_range,
+                &peer_list,
+                &data_key_address,
+                &cfg.get_quorum,
+                false,
+            );
+
+            let mut accumulated_spends = BTreeSet::new();
+
+            // try and accumulate any SpendAttempts
+            if results_count > 1 {
+                info!("For record {pretty_key:?} task {query_id:?}, we have more than one result returned.");
+                // Allow for early bail if we've already seen a split SpendAttempt
+                for (record, _) in result_map.values() {
+                    match get_raw_signed_spends_from_record(record) {
+                        Ok(spends) => {
+                            accumulated_spends.extend(spends);
+                        }
+                        Err(_) => {
+                            continue;
+                        }
+                    }
+                }
+            }
 
-            debug!("Expecting {expected_answers:?} answers for record {pretty_key:?} task {query_id:?}, received {responded_peers} so far");
+            // we have a Double SpendAttempt and should bail early
```

Comment:
>  I don't think it should be kad code's responsibility to judge the content of the records.

Hmm, it's an efficiency thing. If we have a double spend, we could exit earlier instead of waiting for full accumulation. The end result is always the same.

Similarly, we can exit early for Chunks.


It is all very `optimisation`y though. I'll try some runs without this and see where we get to. As you say, maybe we can simplify this out.

Reply:
I think we should wait for full accumulation as if we stop at the first ones we might miss the original one as in the shadowing attack we fixed recently. 
I think low level functions should as much as possible, not take assumptions and do their job in a predictable manner.